baseline/OSPS-DO.yaml

```
@@ -234,9 +238,9 @@ controls:
         applicability:
           - Maturity Level 3
         recommendation: |
-          While a machine-readable OpenEoX file is recommended, this may also be
-          communicated in a SUPPORT.md or beneath a Support header in the
-          primary README.md.
+          In order to communicate the scope and duration of support for security
+          fixes, the project should have a SUPPORT.md or other documentation
+          explaining the project's policy for security updates.
```

I'd love for level 3 to include a machine-readable requirement; possibly something along the lines of:

```suggestion
          In order to communicate the scope and duration of support for security
          fixes, the project must have a machine-readable version support matrix.
          This support matrix should be readable by common aggregation software,
          such as https://endoflife.date/.
```

(I'd love to have a stronger standard to point to, but it feels like level 3 should really support machine-processing of "is my version supported")

---

brainstorming, but... we might also want to add this into the project section of security insights.

https://github.com/ossf/security-insights-spec/blob/main/template-full.yml

---

Yep, agreed. I will open a security insights enhancement proposal

---

https://github.com/ossf/security-insights-spec/issues/112

---

While I agree strongly in the value of machine-readable EoL/support data, I think it makes sense to start with the insights spec change I've proposed above which at least gets us to the point where maintainers can define where this content is documented for human user consumption.

A further insights schema update can also be proposed that introduces a URL to the machine-readable form of that same data. WDYT?

---

I'm slightly concerned about having some parts of security-insights be fast-moving (i.e. needing updates on each release) while other parts are not.  In my experience, it becomes harder to automate _partial_ changes to a file compared with being able to regenerate the whole file (in this case, a version support table).

That's not to say that it can't go into Security Insights, just that I would probably put release-specific information in a separate file from documentation, maintainer, etc data.

---

To state my position on this PR clearly, I'm okay with starting here, but my personal opinion is that if the support documentation isn't machine-consumable, we've missed the real value for _consumers_ of open source.  Ideally, for level 2 or level 3 projects, they should be able to automatically flag when they are producing or deploying an artifact which is outside the upstream support window, and decide how they want to mitigate that risk.

---

> for level 2 or level 3 projects, they should be able to automatically flag when they are producing or deploying an artifact which is outside the upstream support window, and decide how they want to mitigate that risk

This is interesting - I could imagine this type of control becoming `OSPS-QA-02.03` perhaps